sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_users_services_first_seen_v1/query.sql

```diff
@@ -1,136 +1,93 @@
-WITH base AS (
+WITH fxa_events AS (
   SELECT
-    * REPLACE (
-      IF(service IS NULL AND event_type = 'fxa_activity - cert_signed', 'sync', service) AS service
-    )
+    timestamp,
+    user_id,
+    IF(service IS NULL AND event_type = 'fxa_activity - cert_signed', 'sync', service) AS service,
+    os_name,
+    flow_id,
+    event_type,
+    os_version,
+    country,
+    language,
+    utm_term,
+    utm_source,
+    utm_medium,
+    utm_campaign,
+    utm_content,
+    ua_version,
+    ua_browser,
+    entrypoint,
   FROM
-    firefox_accounts.fxa_content_auth_oauth_events
+    `moz-fx-data-shared-prod.firefox_accounts.fxa_content_auth_oauth_events`
+  WHERE DATE(timestamp) = DATE(@submission_date)  -- # TODO: need to verify how long the flow session lasts. Might have to make this window a bit bigger
+  -- # to make sure we don't miss any entries where the flow started just before midnight
 ),
-  -- use a window function to look within each USER and SERVICE for the first value of service, os, and country.
-  -- also, get the first value of flow_id for later use and create a boolean column that is true if the first instance of a service usage includes a registration.
-  -- [kimmy] the variable first_service_timestamp_last is named so because it is actually the last timestamp recorder in the user's first flow,
-  -- NOT the first timestamp in their first flow.
-  -- it's used later on to order by service, so i'm keeping it here and just renaming it.
-first_services AS (
-  SELECT
-    ROW_NUMBER() OVER w1_unframed AS _n,
+
+new_user_service_os_entries AS (
+  SELECT DISTINCT
+    flow_id,
+    timestamp,
     user_id,
     service,
-      -- using mode_last with w1_reversed to get mode_first
-    udf.mode_last(ARRAY_AGG(`timestamp`) OVER w1_reversed) AS first_service_timestamp_last,
-    udf.mode_last(ARRAY_AGG(flow_id) OVER w1_reversed) AS first_service_flow,
-    LOGICAL_OR(IFNULL(event_type = 'fxa_reg - complete', FALSE)) OVER w1_reversed AS did_register
+    os_name,
+    os_version,
+    country,
+    language,
+    utm_term,
+    utm_source,
+    utm_medium,
+    utm_campaign,
+    utm_content,
+    ua_version,
+    ua_browser,
+    entrypoint,
   FROM
-    base
+    fxa_events
   WHERE
-    (
-      (event_type IN ('fxa_login - complete', 'fxa_reg - complete') AND service IS NOT NULL)
-      OR (event_type LIKE 'fxa_activity%')
+    flow_id IS NOT NULL
+    AND ((event_type IN ('fxa_login - complete', 'fxa_reg - complete') AND service IS NOT NULL))
+    AND CONCAT(user_id, service, os_name) NOT IN (
+      SELECT CONCAT(user_id, service, os_name)
+      FROM `moz-fx-data-shared-prod.firefox_accounts_derived.fxa_users_services_first_seen_v1`
```

Comment:
This should only scan prior days, so that we can run backfills:
```suggestion
      FROM `moz-fx-data-shared-prod.firefox_accounts_derived.fxa_users_services_first_seen_v1`
      WHERE first_seen_date < @submission_date
```

Reply:
that's a very good point. Thank you for pointing this out! 🙇